fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1401,6 +1401,13 @@ public Bundle doSearch(String type, String compartment, String compartmentId,
             // Invoke the 'afterSearch' interceptor methods.
             getInterceptorMgr().fireAfterSearchEvent(event);
 
+            // Interceptors might want to change the response bundle, so make sure pick up the new value.
+            // Protect against an interceptor returning something other than a Bundle
```

Comment:
ClassCastException might actually be ok in that case...it would trigger a 500 server error which seems appropriate for an interceptor thats setting a response resource which could not be cast to the expected response object.
It make me wonder if we should have an else here that emits a warning.
Hopefully no one would do that though, so probably doesn't matter much.

Reply:
Yeah, was having similar thoughts on how to handle this. We probably also want to protect against the resource type being changed (e.g. a Patient read returning an Observation). I was also thinking we might want to verify that the resource id wasn't changed (easy for read/vread...a bit harder if we wanted to check everything inside a search/history response Bundle).